tools/__init__.py

```diff
@@ -228,6 +229,12 @@ def _package_str(self):
             ver_str = self.package
         return ver_str
 
+    def is_attempted(self):
+        return self._is_attempted
+
+    def is_installed(self):
```

Comment:
by changing is_installed here, won't we lose the memoization of this return value that we were manually implementing before? so now it has to invoke "conda list" each time? @tomkinsc thoughts